polyfill/test/intl.mjs

```diff
@@ -213,6 +213,431 @@ describe('Intl', () => {
     });
   });
 
+  describe('Non-ISO Calendars', () => {
+    const testChineseData = new Date('2001-02-01T00:00Z').toLocaleString('en-US-u-ca-chinese', {
+      day: 'numeric',
+      month: 'numeric',
+      year: 'numeric',
+      era: 'short',
+      timeZone: 'UTC'
+    });
+    const hasOutdatedChineseIcuData = !testChineseData.endsWith('2001');
+    const itOrSkip = (id) => ((id === 'chinese' || id === 'dangi') && hasOutdatedChineseIcuData ? it.skip : it);
+
+    it('verify that Intl.DateTimeFormat.formatToParts output matches snapshot data', () => {
+      // This test isn't testing Temporal. Instead, it's verifying that the
+      // output of Intl.DateTimeFormat.formatToParts for non-ISO calendars
+      // hasn't changed. There are a number of outstanding bugs in this output
+      // that, when fixed, will break other tests. So this test is a signal that
+      // other tests are broken because the comparison data needs to be updated,
+      // not necessarily because Temporal is broken.
+      // prettier-ignore
+      // eslint-disable-next-line max-len, no-console, brace-style
+      const year2000Content = ['iso8601', 'buddhist', 'chinese', 'coptic', 'dangi', 'ethioaa', 'ethiopic', 'hebrew', 'indian', 'islamic', 'islamic-umalqura', 'islamic-tbla', 'islamic-civil', 'islamic-rgsa', 'islamicc', 'japanese', 'persian', 'roc'].map((id) => `${id}: ${new Date('2000-01-01T00:00Z').toLocaleDateString('en-us', { calendar: id, timeZone: 'UTC' })}`).join('\n');
```

Comment:
I'm working on this section now-- will probably end up with a Node 12, Node 14, and Node 15 variation of this test using different strings to match.

Reply:
BTW, I ended up not using `era: 'short'` here because adding that option changed the output behavior for some but not all calendars. The actual implementation does use `era` and tests cover that usage separately. The purpose of these tests was mostly as a heads-up in case a future Node release contained bug fixes that might break other tests, so that we don't end up in a wild goose chase wondering why previously-working tests are broken. 

I revised the Node 12 output for this test to simply verify that the date calculations are as expected, and it ignores the (missing) eras. I think this is probably fine given that these tests are kinda optional anyways, but holler if you disagree.